tests/integration_tests/datasource_tests.py

```diff
@@ -385,21 +387,30 @@ def my_check(datasource):
         app.config["DATASET_HEALTH_CHECK"] = my_check
         self.login(username="admin")
         tbl = self.get_table(name="birth_names")
-        datasource = ConnectorRegistry.get_datasource("table", tbl.id, db.session)
+        datasource = db.session.query(SqlaTable).filter_by(id=tbl.id).one_or_none()
         assert datasource.health_check_message == "Warning message!"
         app.config["DATASET_HEALTH_CHECK"] = None
 
     def test_get_datasource_failed(self):
+        from superset.datasource.dao import DatasourceDAO
+
         pytest.raises(
-            DatasetNotFoundError,
-            lambda: ConnectorRegistry.get_datasource("table", 9999999, db.session),
+            DatasourceNotFound,
+            lambda: DatasourceDAO.get_datasource(db.session, "table", 9999999),
```

Comment:
These are testing the actual function calls inside the `views/datasource.py` files which call `DatasourceDAO` now